api/src/test/java/org/apache/iceberg/TestIcebergBuild.java

```diff
@@ -37,6 +41,26 @@ public void testFullVersion() {
                 + ")");
   }
 
+  @Test
+  public void testVersionNotUnspecified() {
+    assertThat(IcebergBuild.version()).isNotEqualTo("unspecified");
+  }
+
+  /**
+   * This test is for Source Releases. When we have a source release we use a version.txt file in
+   * the parent directory of this module to actually set the "version" which should be included in
+   * the gradle build properties used by IcebergBuild.
+   */
+  @Test
+  public void testVersionMatchesFile() throws IOException {
+    Path versionPath = Paths.get("../version.txt").toAbsolutePath();
+    assumeThat(java.nio.file.Files.exists(versionPath)).isTrue();
```

Comment:
nit: 

If we set the `project.version` in the system property for tests in `build.gradle`

```
test {
    systemProperty "project.version", project.version
}
```

we can always check 

```
assertThat(IcebergBuild.version())
        .isEqualTo(System.getProperty("project.version"))
```

This way we don't need two testcase and also it won't be a conditional test.